sktime/transformations/series/bootstrapping.py

```diff
@@ -0,0 +1,395 @@
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Bootstrapping methods for time series."""
+
+__author__ = ["ltsaprounis"]
+
+from copy import copy
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+from statsmodels.tsa.api import STL as _STL
+
+from sktime.transformations.base import BaseTransformer
+from sktime.transformations.series.boxcox import BoxCoxTransformer
+
+
+class UnivariateBootsrappingTransformer(BaseTransformer):
+    """Creates a population of similar time series.
+
+    This method utilises a form of bootstrapping to generate a population of
+    similar time series to the input time series [1]_, [2]_.
+
+    First the observed time series is transformed using a Box-Cox transformation to
+    stabilise the variance. Then it's decomposed to seasonal, trend and remainder
+    time series, using the STL algorithm from statsmodels [4]_. We then sample blocks
+    from the remainder time series using the Moving Block Bootstrapping (MBB) method
+    [3]_ to create synthetic remainder series that mimic the autocorrelation patterns
+    of the observed series. Finally these bootstrapped remainders are added to the
+    season and trend components and we use the inverse Box-Cox transform to return
+    a panel of similar time series.
+
```

Comment:
Could you maybe describe here also some examples of good use-cases for this transformer?